anta/tools.py

```diff
@@ -36,6 +38,24 @@ def get_failed_logs(expected_output: dict[Any, Any], actual_output: dict[Any, An
     return "".join(failed_logs)
 
 
+def custom_division(numerator: float, denominator: float) -> int | float:
+    """Get the custom division of numbers.
+
+    Custom division that returns an integer if the result is an integer, otherwise a float.
+
+    Args:
+    ----
+    numerator (float): The numerator.
+    denominator (float): The denominator.
```

Comment:
we rely on type hints for types

```suggestion
    numerator: The numerator.
    denominator: The denominator.
```